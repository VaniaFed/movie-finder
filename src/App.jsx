import React from 'react';
import { v4 } from 'uuid';
import { SearchPage } from 'Components/search-page/';
import { MoviePage } from 'Components/movie-page';
import { Logo } from 'Components/logo';
import { ErrorBoundary } from 'Components/error-boundary';

const movies = [
    {
        title: 'movei1',
        genres: 'action',
        tagline: 'some tagline',
        releaseDate: '2014',
        runtime: 160,
        overview:
            'Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolor quidem non id numquam nulla, porro itaque laborum ut, minima illum quaerat, iusto dignissimos harum quae? Corrupti sit laboriosam optio magnam!',
        posterPath:
            'https://pp.userapi.com/c844521/v844521991/f808b/TCjtAXN8SdA.jpg',
        id: v4()
    },
    {
        title: 'movei1',
        genres: 'action',
        tagline: 'some tagline',
        releaseDate: '2014',
        posterPath:
            'https://pp.userapi.com/c844521/v844521991/f808b/TCjtAXN8SdA.jpg',
        id: v4()
    },
    {
        title: 'movei1',
        genres: 'action',
        tagline: 'some tagline',
        releaseDate: '2014',
        posterPath:
            'https://pp.userapi.com/c844521/v844521991/f808b/TCjtAXN8SdA.jpg',
        id: v4()
    }
];
export const App = () => {
    return (
        <ErrorBoundary>
            <div className="container">
                <SearchPage />
                <MoviePage movie={movies[0]} moviesWithTheSameGenre={movies} />
                <footer>
                    <Logo>netfixroulette</Logo>
                </footer>
            </div>
        </ErrorBoundary>
    );
};
