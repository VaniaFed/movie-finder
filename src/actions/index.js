import {
    fetchMovieByIdRequest,
    fetchMovieByIdSuccess,
    fetchMovieByIdError
} from 'actions/movie-by-id';
import {
    fetchMoviesByDataRequest,
    fetchMoviesByDataSuccess,
    fetchMoviesByDataError
} from 'actions/movies-by-data';

import { searchFilter } from 'actions/search-filter';
import { sortFilter } from 'actions/sort-filter';

export const actions = {
    fetchMovieByIdRequest,
    fetchMovieByIdSuccess,
    fetchMovieByIdError,

    fetchMoviesByDataRequest,
    fetchMoviesByDataSuccess,
    fetchMoviesByDataError,

    searchFilter,

    sortFilter
};
