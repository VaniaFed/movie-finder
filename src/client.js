import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { RestLink } from 'apollo-link-rest';

const restLink = new RestLink({
    uri: 'http://react-cdp-api.herokuapp.com/movies'
});
const cache = new InMemoryCache({
    dataIdFromObject: movie => movie.id || null
});

export const client = new ApolloClient({
    link: restLink,
    cache
});
