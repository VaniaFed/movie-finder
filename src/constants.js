export const FETCH_MOVIES_BY_GENRE_REQUEST = 'FETCH_MOVIES_BY_GENRE_REQUEST';
export const FETCH_MOVIES_BY_GENRE_SUCCESS = 'FETCH_MOVIES_BY_GENRE_SUCCESS';
export const FETCH_MOVIES_BY_GENRE_ERROR = 'FETCH_MOVIES_BY_GENRE_ERROR';

export const FETCH_MOVIE_BY_ID_REQUEST = 'FETCH_MOVIES_BY_ID_REQUEST';
export const FETCH_MOVIE_BY_ID_SUCCESS = 'FETCH_MOVIES_BY_ID_SUCCESS';
export const FETCH_MOVIE_BY_ID_ERROR = 'FETCH_MOVIES_BY_ID_ERROR';

export const FETCH_MOVIES_BY_DATA_REQUEST = 'FETCH_MOVIES_BY_DATA_REQUEST';
export const FETCH_MOVIES_BY_DATA_SUCCESS = 'FETCH_MOVIES_BY_DATA_SUCCESS';
export const FETCH_MOVIES_BY_DATA_ERROR = 'FETCH_MOVIES_BY_DATA_ERROR';
