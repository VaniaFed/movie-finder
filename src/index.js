import { App } from './components/App';
import React from 'react';
import { render } from 'react-dom';

import "./styles.sass";

render(<App />,
    document.querySelector('#app')
)
