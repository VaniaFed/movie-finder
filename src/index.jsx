import React from 'react';
import { render } from 'react-dom';
import { App } from 'Components/App';

import './global.sass';

render(<App />, document.querySelector('#app'));
