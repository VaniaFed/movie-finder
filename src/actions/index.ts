import {
    fetchMovieByIdRequest,
    fetchMovieByIdSuccess,
    fetchMovieByIdError
} from 'actions/movie-by-id';
import {
    fetchMoviesByDataRequest,
    fetchMoviesByDataSuccess,
    fetchMoviesByDataError
} from 'actions/movies-by-data';
import {
    fetchMoviesByGenreRequest,
    fetchMoviesByGenreSuccess,
    fetchMoviesByGenreError
} from 'actions/movies-by-genre';
import { setSearchBy } from 'actions/set-search-by';
import { setSortBy } from 'actions/set-sort-by';
import { setSearch } from 'actions/set-search';

export const actions = {
    fetchMovieByIdRequest,
    fetchMovieByIdSuccess,
    fetchMovieByIdError,

    fetchMoviesByDataRequest,
    fetchMoviesByDataSuccess,
    fetchMoviesByDataError,

    fetchMoviesByGenreRequest,
    fetchMoviesByGenreSuccess,
    fetchMoviesByGenreError,

    setSearchBy,

    setSortBy,

    setSearch
};
