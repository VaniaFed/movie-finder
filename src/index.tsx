import 'src/client';
import React from 'react';
import { hydrate } from 'react-dom';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import { App } from 'containers/App';
import { configureStore } from 'src/store/';

import './global.scss';

const store = configureStore();
hydrate(
    <Router>
        <Provider store={store}>
            <App />
        </Provider>
    </Router>,
    document.getElementById('app')
);
