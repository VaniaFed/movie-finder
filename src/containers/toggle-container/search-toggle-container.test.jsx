import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { configureStore } from 'src/configure-store';
import { SearchToggleContainer } from './index';

const store = configureStore();
test('<SearchToggleContainer />', () => {
    const sortByData = [
        { key: 'release_date', text: 'Release date', onClick: f => f },
        { key: 'rating', text: 'Rating', onClick: f => f }
    ];
    const tree = renderer
        .create(
            <Provider store={store}>
                <SearchToggleContainer
                    name="search"
                    background
                    data={sortByData}
                />
            </Provider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
