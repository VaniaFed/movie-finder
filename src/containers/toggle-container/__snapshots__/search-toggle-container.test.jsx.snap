// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchToggleContainer /> 1`] = `
<div>
  <label
    className="toggle-button"
    onClick={[Function]}
  >
    <input
      className="toggle-button__input toggle-button__input--checked"
      defaultChecked="checked"
      name="search"
      type="radio"
    />
    <span
      className="toggle-button__text toggle-button__text--checked"
    >
      Release date
    </span>
  </label>
  <label
    className="toggle-button"
    onClick={[Function]}
  >
    <input
      className="toggle-button__input"
      defaultChecked={false}
      name="search"
      type="radio"
    />
    <span
      className="toggle-button__text"
    >
      Rating
    </span>
  </label>
</div>
`;

exports[`<SearchToggleContainer /> should be rendered 1`] = `
<div>
  <label
    className="toggle-button"
    onClick={[Function]}
  >
    <input
      className="toggle-button__input toggle-button__input--checked"
      defaultChecked="checked"
      name="search"
      type="radio"
    />
    <span
      className="toggle-button__text toggle-button__text--checked"
    >
      Release date
    </span>
  </label>
  <label
    className="toggle-button"
    onClick={[Function]}
  >
    <input
      className="toggle-button__input"
      defaultChecked={false}
      name="search"
      type="radio"
    />
    <span
      className="toggle-button__text"
    >
      Rating
    </span>
  </label>
</div>
`;
